### ISSUE TITLE
How to configure the dds to use tcp

### ISSUE THREAD

**Issue by MatiasVara** [eclipse-cyclonedds/cyclonedds#661]
Hello, I would like to configure the DDS to use TCP in the HelloWorld example. For the moment, I have just changed the transport layer in the .xml to tcp but it does not seem working. What am I missing? Thanks.

**Comment by MatiasVara** [comment]
It works by adding the following lines in the Publisher:
```
<TCP>
<Port>3000</Port>
</TCP>
```
and the following lines in the subscriber:
```
<Discovery>
  <Peers>
     <Peer Address="IPofPublisher:3000"/>
  </Peers>
</Discovery>
```
Both with tcp transport layer.

**Comment by eboasson** [comment]
Yes, that's what you need to do. TCP's design around "clients" and "servers" is a bit of a bad match for a peer-to-peer system, so you have to plan your set up a bit, especially also because you can't have multiple processes listening on the same TCP port on a single machine.

You might be interested in https://github.com/eclipse-zenoh/zenoh-plugin-dds. It is still somewhat limited in what it can do, but it'll grow to provide full DDS discovery and routing services.

**Comment by MatiasVara** [comment]
Thanks, I will take a look. I am only interested on the tcp layer because I am working on a PoC to add a new transport layer. I am building the PoC by modifying the tcp layer.
